static/src/javascripts/projects/commercial/modules/dfp/Advert.ts

```diff
@@ -119,62 +92,35 @@ class Advert {
 
 		this.slot = slotDefinition.slot;
 		this.whenSlotReady = slotDefinition.slotReady;
-
-		this.whenLoaded = new Promise((resolve: Resolver) => {
-			this.whenLoadedResolver = resolve;
-		}).then((isLoaded: boolean): boolean => {
-			this.isLoaded = isLoaded;
-			return isLoaded;
-		});
-
-		this.whenRendered = new Promise((resolve: Resolver) => {
-			this.whenRenderedResolver = resolve;
-		}).then((isRendered) => {
-			this.isRendered = isRendered;
-			return isRendered;
-		});
-	}
-
-	static filterClasses = (
-		oldClasses: string[],
-		newClasses: string[],
-	): string[] =>
-		oldClasses.filter((oldClass) => !newClasses.includes(oldClass));
-
-	startLoading(): void {
-		this.isLoading = true;
-		this.timings.startLoading = window.performance.now();
-	}
-
-	stopLoading(isLoaded: boolean): void {
-		this.isLoading = false;
-		if (this.whenLoadedResolver) {
-			this.whenLoadedResolver(isLoaded);
-		}
-		this.timings.stopLoading = window.performance.now();
-	}
-
-	startRendering(): void {
-		this.isRendering = true;
-		this.timings.startRendering = window.performance.now();
 	}
 
-	stopRendering(isRendered: boolean): void {
-		this.isRendering = false;
-		if (this.whenRenderedResolver) {
-			this.whenRenderedResolver(isRendered);
-		}
+	/**
+	 * Call this method once the ad has been rendered, it will set the
+	 * `isRendered` flag to true, which is used to determine whether to load
+	 * or refresh the ad
+	 *
+	 * @param isRendered was an advert rendered
+	 */
+	finishedRendering(isRendered: boolean): void {
+		this.isRendered = isRendered;
 	}
 
+	/**
+	 * Update the "extra" classes for this slot e.g. `ad-slot--outstream`, so that the main one's
+	 * like `ad-slot` etc. are not affected
+	 *
+	 * @param newClasses An array of classes to set on the slot
+	 **/
 	updateExtraSlotClasses(...newClasses: string[]): void {
-		const classesToRemove = Advert.filterClasses(
-			this.extraNodeClasses,
-			newClasses,
+		const classesToRemove = this.extraNodeClasses.filter(
+			(c) => !newClasses.includes(c),
 		);
-		// IE11 does not support multiple arguments to classList.add/remove so do these one-by-ones
-		classesToRemove.forEach((cls) => this.node.classList.remove(cls));
-		newClasses.forEach((cls) => this.node.classList.add(cls));
-		this.extraNodeClasses = newClasses;
+
+		void fastdom.mutate(() => {
+			this.node.classList.remove(...classesToRemove);
+			this.node.classList.add(...newClasses);
+			this.extraNodeClasses = newClasses;
```

Comment:
This line doesn't need to be in the `fastdom.mutate()` ?

Reply:
Is changing classes not DOM mutation?

Reply:
Oh I was referring to line #122 - isn't that just updating a member variable?

Reply:
Oh sorry, I miss-understood, I though it might be possible for the the line to executed first if it was outside the fastdom callback?

Reply:
I think it needs to exist as a way for any `extra` classes to be removed without affecting the base classes, so calling it with no arguments will remove the extra class but leave `ad-slot` etc. intact.